presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -142,6 +142,18 @@ std::shared_ptr<connector::ColumnHandle> toColumnHandle(
         toRequiredSubfields(hiveColumn->requiredSubfields));
   }
 
+  if (auto icebergColumn =
+          dynamic_cast<const protocol::IcebergColumnHandle*>(column)) {
+    // TODO(imjalpreet): Modify 'hiveType' argument of the 'HiveColumnHandle'
```

Comment:
I haven't looked into this change in depth yet but yes it would require additional changes on the coordinator side before we can implement the changes here.